helix-vcs/src/git.rs

```diff
@@ -61,7 +65,30 @@ impl DiffProvider for Git {
         let file_oid = find_file_in_commit(&repo, &head, file)?;
 
         let file_object = repo.find_object(file_oid).ok()?;
-        Some(file_object.detach().data)
+        let mut data = file_object.detach().data;
+        // convert LF to CRLF if configured to avoid showing every line as changed
+        if repo
+            .config_snapshot()
```

Comment:
Note that this will not update throughout the lifetime of the `Repository`. Soon there should will be a `config()` counterpart which will assure all involved configuration is also uptodate.

In this case, I don't think it's an issue as this setting is very stable and usually configured by the platform installation itself to apply for all repositories on the system.

Reply:
Repos are not cached right now anyways (we just opened that repo a few lines before) and there are not automatic reloads. Therefore if you want to update your gutter to reflect any on-disk changes you need to call `:reload` or `:reload-all` anyways which will always fully reload the repo from disk. This will becomes more of an issue once repo caching is enabled. But then your point apllies that this setting is rarely changed. In fact it's often set for windows git installations by default